File: source/aoBlogs2/Models/BlogCategoriesModel.vb


Option Explicit On
Option Strict On

Imports System
Imports System.Collections.Generic
Imports System.Text
Imports Contensive.BaseClasses

Namespace Models
    Public Class BlogCategoriesModel        '<------ set set model Name and everywhere that matches this string
        Inherits DbModel
        '
        '====================================================================================================
        '-- const
        Public Const contentName As String = "Blog Categories"      '<------ set content name
        Public Const contentTableName As String = "BlogCategories"   '<------ set to tablename for the primary content (used for cache names)
        Private Shadows Const contentDataSource As String = "default"             '<------ set to datasource if not default
        '
        '====================================================================================================
        ' -- instance properties
        Public Property UserBlocking As Boolean

    End Class
End Namespace

File: source/aoBlogs2/Models/Db/BlogCategoriesModel.cs


using Contensive.Models.Db;

namespace Contensive.Blog.Models {
    public class BlogCategoriesModel : Contensive.Models.Db.DbBaseModel {
        // 
        // ====================================================================================================
        //
        public static DbBaseTableMetadataModel tableMetadata { get; private set; } = new DbBaseTableMetadataModel("Blog Categories", "BlogCategories", "default", false);
        //
        // ====================================================================================================
        // -- instance properties
        public bool UserBlocking { get; set; }

    }
}